src/backend/tests/conftest.py

```diff
@@ -204,3 +205,19 @@ def mock_available_model_deployments(request):
 
     with patch.dict(AVAILABLE_MODEL_DEPLOYMENTS, MOCKED_DEPLOYMENTS) as mock:
         yield mock
+
+
+@pytest.fixture
+def mock_compass_settings():
+    with patch("backend.services.file.Settings") as MockSettings:
+        mock_settings = MockSettings.return_value
+        mock_settings.feature_flags.use_compass_file_storage = os.getenv(
+            "ENABLE_COMPASS_FILE_STORAGE", False
+        )
+        mock_settings.tools.compass.api_url = os.getenv("COHERE_COMPASS_API_URL")
+        mock_settings.tools.compass.api_parser_url = os.getenv(
+            "COHERE_COMPASS_API_PARSER_URL"
+        )
+        mock_settings.tools.compass.username = os.getenv("COHERE_COMPASS_USERNAME")
+        mock_settings.tools.compass.password = os.getenv("COHERE_COMPASS_PASSWORD")
+        yield mock_settings
```

Comment:
Seems like when `Settings()` is called in the code during a pytest there may be some issues with its instantiation.

This overrides the results when it is called for compass settings. Also might be useful if we ever want to run the compass tests with on a test instance with different credentials, etc.

Though, after fixing and running tests, they are very slow. We've faced this issue with testing on the test DB. Perhaps time to start investing into mocking instead of using these direct calls on storage. For now this will suffice.